dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/util/MathUtils.java

```diff
@@ -71,6 +72,8 @@
 
     private static final Pattern ZERO_PATTERN = Pattern.compile( "^0(\\.0*)?$" );
 
+    private static final Random RANDOM = ThreadLocalRandom.current();
+
```

Comment:
Just wondering if this is a great idea, as a class scope variable means that all threads will use the same instance.

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html

From the javadoc it seems the recommended usage is:

```
ThreadLocalRandom.current().nextX(...) 
```
I.e. `ThreadLocalRandom.current()` returns the current thread's ThreadLocalRandom.


[current](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html#current--)()
Returns the current thread's ThreadLocalRandom.

Reply:
I think you are right, I'll remedy this right away!